src/string_id.h

```diff
@@ -39,6 +42,25 @@ class int_id;
  * please define it in type_id.h, a central light-weight header that defines all ids
  * people might want to use.  This prevents duplicate definitions in many
  * files.
+ *
+ * Notes on usage and performance (comparison between ids, ::obj lookup,
+ *    assuming default implementation of generic_factory is used):
+ *
+ * `int_id` is fastest, but it can't be reused after game reload. Depending on the loaded
+ *    combination of mods `int_id` might point to a different entity and there is no way to tell.
+ *    Because of this NEVER define static int ids.
```

Comment:
I think we do currently have some static `int_id`s (with special handling so that they get reset on game load).  Now we can probably get rid of those, which is cool, and create a `clang-tidy` check to ensure no static `int_id`s are created. (not in this PR, though).